code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -566,3 +587,94 @@
 		else
 			qdel(G)
+
+/mob/living/carbon/human/proc/empath()
+	set category = "Abilities"
+	set name = "Empath Reading"
+	set desc = "Glean from the minds of others."
+
+
+	var/obj/item/organ/external/rhand = src.get_organ("r_hand")
+	var/obj/item/organ/external/lhand = src.get_organ("l_hand")
+	if((!rhand || !rhand.is_usable()) && (!lhand || !lhand.is_usable()))
+		src << "<span class='warning'>You can't use your power without the ability to use your hands!</span>"
+		return FALSE
+	if(stat || paralysis || stunned || weakened ||  restrained())
+		src << "<span class='warning'>You can't use your power while unable to move your hands to your head!</span>"
+		return FALSE
+	if(last_special > world.time)
+		src << "<span class='notice'>Your mind requires rest!</span>"
+		return FALSE
+
+	last_special = world.time + 70
+
+	visible_message("<span class='notice'>[src] touches their fingers to their temple.</span>")
+
+	var/list/targets = list()
+	for(var/mob/living/carbon/human/M in view(7, src))
+		targets += M
+	var/mob/living/carbon/human/target = null
+
+	target = input("Select a creature!", "Read a creature", null, null) as null|anything in targets
+
+
+	if(!target)
+		return
+
+	if (!target || !target.loc) //Either chose not to, or the mob was caught by qdel
+		return 1
+
+	if (isSynthetic(target))
+		to_chat(src, "<span class='warning'>This can only be used on carbon beings.</span>")
+		return 1
+
+	if (istype(target, /mob/abstract/observer || target.stat == DEAD))
+		to_chat(src, "Not even a [src.species.name] can read the minds of the dead.")
+		return 1
+
+	if (target.stat == UNCONSCIOUS)
+		to_chat(src, "<span class='warning'>[target.name] is unconscious, and their thoughts blank.</span>")
+		return 1
+
+	if(get_dist(get_turf(target), get_turf(src)) > 7)
+		to_chat(src, "<span class='warning'>[target] is too far for your mind to grasp!</span>")
+		return FALSE
+
+	to_chat(src, "<span class='notice'><b>The inner mind of [target.name] conveys</b></span>")
+	var/list/randomthoughts = list("what to have for lunch","the future","the past","money", "incredible rage", "world domination",
+	"their hair","what to do next","their job","space","amusing things","sad things","a lover",
+	"annoying things","happy little trees","something incoherent","something they did wrong",
+	"the chef", "engineering", "medical", "cargo", "command", "the bar","science","something spooky","something lewd","odd things",
+	"tajara", "skrell", "unathi", "humans", "vaurca", "monkeys", "security officers", "robots",
+	"dumb things","lighting things on fire","lighting themselves on fire","blowing things up",
+	"blowing themeselves up", "you", "shoujo", "karate lessons", "when the shuttle will arrive", "mischief", "boredom", "vodka")
+	var/thoughts = "thinking about [target.feel ? target.feel : pick(randomthoughts)]"
+	if (fire_stacks > 0 && on_fire)
+		thoughts = "preoccupied with the fire"
+
+	switch(target.a_intent)
```

Comment:
_Person A beats person B to death with a crowbar. Skrell uses mindread on him. "Their mood conveys: A sense of intent to help"_
Are we going to enforce the proper setting of the intent depending on the situations now ?
Imho that implementation is suboptimal and will yield very different results.

Also: "They mood conveys: A sense of the intent to grab"

Reply:
what would you recommend

Reply:
there's plenty of ways to go about it:

you could check recent attack logs on the player being mind-scanned, for example.